### PR TITLE
feat: add structured data for SEO

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendu@apache.org>

Related to #980

Adds structured data for the homepage. This is an initial setup and in the future we can automatically generate this data on every page in the site.

Screenshot 2023-02-03 at 11 16 30 AM

### CODE REVIEW

**Comment by pottekkat** [comment]
Also requesting a review from @nfrankel

**Comment by pottekkat** [comment]
@SkyeYoung Please let me know if I need to move the schema definition to some place else.

File: website/src/pages/index.tsx [hunk 1]
```diff
@@ -44,6 +44,33 @@ const ThemeResetComponent = () => {
...
+    "https://raw.githubusercontent.com/apache/apisix/master/logos/apisix-white-bg.jpg",
+  sameAs: [
+    "https://twitter.com/ApacheAPISIX",
+    "https://github.com/apache/apisix",
+    "https://www.youtube.com/@apacheapisix",
+    "https://www.linkedin.com/company/apache-apisix/",
+  ],
+  isPartOf: "Apache Software Foundation",
+  maintainer: "Apache Software Foundation",
+  contributor: [
```
Comment (GeekEmily) [inline website/src/pages/index.tsx]: What are the criteria to pick the contributors?

**Comment by SkyeYoung** [comment]
> @SkyeYoung Please let me know if I need to move the schema definition to some place else.

No need I think. 🤔 But we need to improve some of the content.

File: website/src/pages/index.tsx (hunk 1 above)
Comment (pottekkat) [reply website/src/pages/index.tsx]: I just went to orbit and found top organizations that contribute to APISIX. Should we do it?

Comment (nfrankel) [reply website/src/pages/index.tsx]: You need to have approval of some representative of the organizations for that.

Comment (pottekkat) [reply website/src/pages/index.tsx]: Removed the section.

**Comment by pottekkat** [comment]
> No need I think. 🤔 But we need to improve some of the content.

Could you share what we can improve?

**Comment by SkyeYoung** [comment]
> Could you share what we can improve?

Sorry, I'm late. My thoughts are similar to what keer said, and I think it's good now that it's been revised.

**Review by juzhiyuan — APPROVED** [review]
Just to confirm: this only adds Schema to the homepage?

**Comment by pottekkat** [comment]
> Just to confirm: this only adds Schema to the homepage?

Yes. It only does that. But AFAIK it will help with our goal to get a knowledge panel. In future, we can generate schema automatically for each page from the metadata. I have to spend more time looking into how I can achieve this or someone else can take it up.

File: website/src/pages/index.tsx [hunk 2]
```diff
@@ -64,6 +85,11 @@ const Index: FC = () => (
         name="og:description"
         content="Apache APISIX is a dynamic, real-time, high-performance Cloud-Native API gateway, based on the Nginx library and etcd."
       />
+
+      <script type="application/ld+json">
```
Comment (nfrankel) [inline website/src/pages/index.tsx:89]: Why don't you put the JSON directly here?

**Comment by nfrankel** [comment]
[Google Structured Data Testing Tool](https://search.google.com/test/rich-results) isn't happy about some missing fields:

Screenshot 2023-02-06 at 09 29 42

File: website/src/pages/index.tsx (hunk 2 above)
Comment (pottekkat) [reply website/src/pages/index.tsx:89]: It did not work when I directly added the JSON. I looked at an example implementation, and they used a similar approach.

**Comment by nfrankel** [comment]
I seem to remember Google complaining about missing field why `Blog` was easier to start with (cf. https://github.com/apache/apisix-website/issues/980#issuecomment-1156033243).

**Comment by pottekkat** [comment]
@nfrankel Should we add `applicationCategory` and `operatingSystem`? Another alternative is to use https://schema.org/WebSite

Which way should we go?

**Comment by nfrankel** [comment]
@navendu-pottekkat As I mentioned, it should be easier to start with `Blog`

**Pull request closed without merging**